pkg/option/config.go

```diff
@@ -3060,7 +3049,6 @@ func (c *DaemonConfig) Populate(logger *slog.Logger, vp *viper.Viper) {
 		c.IdentityRestoreGracePeriod = defaults.IdentityRestoreGracePeriodKvstore
 	}
 
-	c.LoadBalancerProtocolDifferentiation = vp.GetBool(LoadBalancerProtocolDifferentiation)
```

Comment:
Also, the flag needs to be removed from the helm config map: [install/kubernetes/cilium/templates/cilium-configmap.yaml ](https://github.com/cilium/cilium/blob/main/install/kubernetes/cilium/templates/cilium-configmap.yaml#L839-L841)